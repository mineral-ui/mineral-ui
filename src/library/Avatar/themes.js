/* @flow */
import type { AvatarThemeFn } from './types';

export const avatarTheme: AvatarThemeFn = (baseTheme) => ({
  Avatar_fontSize_small: baseTheme.fontSize_mouse,
  Avatar_fontSize_medium: baseTheme.fontSize_ui,
  Avatar_fontSize_large: baseTheme.h4_fontSize,
  Avatar_fontSize_jumbo: baseTheme.h4_fontSize,
  Avatar_fontWeight: baseTheme.fontWeight_bold,
  Avatar_size_small: baseTheme.size_small,
  Avatar_size_medium: baseTheme.size_medium,
  Avatar_size_large: baseTheme.size_large,
  Avatar_size_jumbo: baseTheme.size_jumbo,

  ...baseTheme
});
