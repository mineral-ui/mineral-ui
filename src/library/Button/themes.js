/* @flow */
import { pxToEm } from '../styles';

import type { ButtonThemeFn } from './types';

// prettier-ignore
export const buttonTheme: ButtonThemeFn = (baseTheme) => ({
  Button_backgroundColor: baseTheme.backgroundColor,
  Button_backgroundColor_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_focus: baseTheme.backgroundColor_focus,
  Button_backgroundColor_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_minimal_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_minimal_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_primary: baseTheme.backgroundColor_themePrimary,
  Button_backgroundColor_primary_active: baseTheme.backgroundColor_themePrimary_active,
  Button_backgroundColor_primary_focus: baseTheme.backgroundColor_themePrimary_focus,
  Button_backgroundColor_primary_hover: baseTheme.backgroundColor_themePrimary_hover,
  Button_borderColor: baseTheme.borderColor,
  Button_borderColor_active: baseTheme.borderColor_theme_active,
  Button_borderColor_focus: baseTheme.borderColor_theme_focus,
  Button_borderColor_hover: baseTheme.borderColor_theme_hover,
  Button_borderRadius: baseTheme.borderRadius_3,
  Button_borderWidth: 1, // px,
  Button_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${baseTheme.borderColor_theme_focus}`,
  Button_color: baseTheme.color_theme,
  Button_color_minimal: baseTheme.color_theme,
  Button_color_primary: baseTheme.color_themePrimary,
  Button_fontWeight: baseTheme.fontWeight_semiBold,
  Button_paddingHorizontal: baseTheme.space_inset_sm_plus,
  Button_paddingIconOnly_small: pxToEm(3),
  Button_paddingIconOnly_medium: pxToEm(7),
  Button_paddingIconOnly_large: pxToEm(7),
  Button_paddingIconOnly_jumbo: pxToEm(13),
  Button_height_small: baseTheme.size_small,
  Button_height_medium: baseTheme.button_size_medium,
  Button_height_large: baseTheme.button_size_large,
  Button_height_jumbo: baseTheme.button_size_jumbo,

  ButtonContent_fontSize: baseTheme.fontSize_ui_sm,
  ButtonContent_fontSize_small: pxToEm(12),

  ButtonIcon_color: baseTheme.icon_color_theme,
  ButtonIcon_margin: baseTheme.space_inset_sm,

  ...baseTheme
});
