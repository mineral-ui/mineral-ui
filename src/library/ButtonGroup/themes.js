/* @flow */
import type { ButtonGroupThemeFn } from './types';

export const buttonGroupTheme: ButtonGroupThemeFn = (baseTheme) => ({
  ButtonGroupButton_backgroundColor_checkedDisabled: baseTheme.color_gray_40,
  ButtonGroupButton_border_disabled: `solid 1px ${baseTheme.borderColor}`,
  ButtonGroupButton_borderColor_active: baseTheme.borderColor_theme_active,
  ButtonGroupButton_borderColor_hover: baseTheme.borderColor_theme_hover,
  ButtonGroupButton_borderStartColor: baseTheme.borderColor,
  ButtonGroupButton_borderStartColor_checked: 'currentcolor',
  ButtonGroupButton_color_checkedDisabled: baseTheme.color_gray_60,

  ...baseTheme
});
