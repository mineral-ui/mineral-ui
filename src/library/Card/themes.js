/* @flow */
import { pxToEm } from '../styles';

import type {
  CardThemeFn,
  CardActionsThemeFn,
  CardBlockThemeFn,
  CardDividerThemeFn,
  CardFooterThemeFn,
  CardStatusThemeFn,
  CardTitleThemeFn
} from './types';

// prettier-ignore
export const cardTheme: CardThemeFn = (baseTheme) => ({
  Card_backgroundColor: baseTheme.panel_backgroundColor,
  Card_borderColor: baseTheme.panel_borderColor,
  Card_borderRadius: baseTheme.borderRadius_1,
  Card_boxShadow: baseTheme.boxShadow_2,
  Card_boxShadow_focus: `0 0 0 2px ${baseTheme.borderColor_theme_focus}, ${baseTheme.boxShadow_1}`,

  CardRow_marginVertical: baseTheme.space_stack_md,
  CardRow_marginVerticalLast: baseTheme.space_stack_lg,
  CardRow_paddingHorizontal: baseTheme.space_inset_md,

  ...baseTheme
});

export const cardActionsTheme: CardActionsThemeFn = (baseTheme) => ({
  CardActionsAction_spaceInline: baseTheme.space_inline_sm,

  ...baseTheme
});

export const cardBlockTheme: CardBlockThemeFn = (baseTheme) => ({
  CardBlock_fontSize: baseTheme.fontSize_ui,
  CardBlock_lineHeight: baseTheme.lineHeight_prose,

  ...baseTheme
});

export const cardDividerTheme: CardDividerThemeFn = (baseTheme) => ({
  CardDivider_borderColor: baseTheme.borderColor,
  CardDivider_borderWidth: '1px',

  ...baseTheme
});

export const cardFooterTheme: CardFooterThemeFn = (baseTheme) => ({
  CardFooter_backgroundColor: baseTheme.well_backgroundColor,
  CardFooter_borderColor: baseTheme.borderColor,

  CardFooterRow_marginVertical: baseTheme.space_stack_sm,
  CardFooterRow_marginVerticalLast: baseTheme.space_stack_md,

  CardFooterTitle_color: baseTheme.h5_color,
  CardFooterTitle_fontSize: baseTheme.h5_fontSize,
  CardFooterTitle_fontWeight: baseTheme.h5_fontWeight,

  ...baseTheme
});

export const cardStatusTheme: CardStatusThemeFn = (baseTheme) => ({
  CardStatus_fontSize: baseTheme.fontSize_ui,
  CardStatus_fontWeight: baseTheme.fontWeight_regular,

  CardStatusIcon_margin: baseTheme.space_inline_sm,
  CardStatusIcon_size: pxToEm(12),

  ...baseTheme
});

export const cardTitleTheme: CardTitleThemeFn = (baseTheme) => ({
  CardTitle_color: baseTheme.h4_color,
  CardTitle_fontSize: baseTheme.h4_fontSize,
  CardTitle_fontWeight: baseTheme.h4_fontWeight,

  CardTitleAvatar_margin: baseTheme.space_inline_sm,
  CardTitleAvatarSize: baseTheme.size_small,
  CardTitleAvatarSize_large: baseTheme.size_large,

  CardTitleIcon_margin: baseTheme.space_inline_sm,

  CardTitleSecondaryText_color: baseTheme.color_mouse,
  CardTitleSecondaryText_fontSize: baseTheme.fontSize_mouse,
  CardTitleSecondaryText_fontWeight: baseTheme.fontWeight_regular,

  CardSubtitle_color: baseTheme.color_mouse,
  CardSubtitle_fontSize: baseTheme.fontSize_mouse,
  CardSubtitle_fontWeight: baseTheme.fontWeight_regular,
  CardSubtitle_marginTop: baseTheme.space_stack_sm,

  ...baseTheme
});
