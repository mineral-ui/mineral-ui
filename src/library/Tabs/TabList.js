/* @flow */
import React, { Children, Component } from 'react';
import debounce from 'lodash.debounce';
import EventListener from '../EventListener';
import IconChevronLeft from '../Icon/IconChevronLeft';
import IconChevronRight from '../Icon/IconChevronRight';
import IconExpandLess from '../Icon/IconExpandLess';
import IconExpandMore from '../Icon/IconExpandMore';
import {
  TabListRoot as Root,
  TabListIncrementButton as IncrementButton,
  TabListInner as Inner,
  TabListList as List
} from './styled';
import { POSITION } from './constants';

import type { TabListDefaultProps, TabListProps, TabListState } from './types';

export default class TabList extends Component<TabListProps, TabListState> {
  static displayName = 'TabList';

  static defaultProps: TabListDefaultProps = {
    position: POSITION.top
  };

  state = {
    scrollable: false
  };

  list: ?HTMLElement;

  root: ?HTMLElement;

  componentDidMount() {
    this.updateScrollable();
  }

  componentDidUpdate(prevProps: TabListProps) {
    if (prevProps !== this.props) {
      this.updateScrollable();
    }
  }

  render() {
    const {
      align,
      children,
      onIncrement,
      position,
      role,
      vertical,
      ...restProps
    } = this.props;
    const { scrollable } = this.state;
    const rootProps = {
      align,
      innerRef: this.setRootRef,
      vertical,
      ...restProps,
      'aria-label': undefined
    };
    const listProps = {
      align,
      'aria-label': restProps['aria-label'],
      count: Children.count(children),
      innerRef: this.setListRef,
      role,
      vertical
    };

    let content = <List {...listProps}>{children}</List>;
    if (scrollable) {
      const innerProps = {
        position,
        scrollX: !vertical,
        scrollY: vertical,
        vertical
      };
      content = [
        <IncrementButton
          key="start"
          icon={vertical ? <IconExpandLess /> : <IconChevronLeft />}
          onClick={(event) => {
            onIncrement && onIncrement('ArrowLeft', event);
          }}
        />,
        <Inner key="inner" {...innerProps}>
          {content}
        </Inner>,
        <IncrementButton
          key="end"
          icon={vertical ? <IconExpandMore /> : <IconChevronRight />}
          onClick={(event) => {
            onIncrement && onIncrement('ArrowRight', event);
          }}
        />
      ];
    }

    return (
      <Root {...rootProps}>
        {content}
        <EventListener
          listeners={[
            {
              event: 'resize',
              handler: this.handleWindowResize,
              target: 'window'
            }
          ]}
        />
      </Root>
    );
  }

  setListRef = (node: ?HTMLElement) => {
    this.list = node;
  };

  setRootRef = (node: ?HTMLElement) => {
    this.root = node;
  };

  updateScrollable = () => {
    const { vertical } = this.props;
    const listNode = this.list;
    const rootNode = this.root;

    if (listNode && rootNode) {
      const dimension = vertical ? 'Height' : 'Width';
      const scrollable = Boolean(
        // $FlowFixMe - indexer property is missing in HTMLElement
        listNode[`scroll${dimension}`] > rootNode[`client${dimension}`]
      );
      if (this.state.scrollable !== scrollable) {
        this.setState({
          scrollable
        });
      }
    }
  };

  handleWindowResize = debounce(this.updateScrollable, 100);
}
