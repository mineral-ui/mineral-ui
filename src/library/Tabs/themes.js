/* @flow */
import type {
  TabsThemeFn,
  TabThemeFn,
  TabListThemeFn,
  TabPanelThemeFn
} from './types';

export const tabsTheme: TabsThemeFn = (baseTheme) => ({
  ...tabListTheme(baseTheme),
  ...tabPanelTheme(baseTheme),
  ...baseTheme
});

export const tabTheme: TabThemeFn = (baseTheme) => ({
  Tab_backgroundColor_selected: baseTheme.backgroundColor_theme_selected,
  Tab_borderColor_focus: baseTheme.borderColor_theme_focus,
  Tab_borderWidth_focus: 1,
  Tab_color: baseTheme.color_gray_80,
  Tab_color_selected: baseTheme.color_theme_70,
  Tab_color_selectedHover: baseTheme.color_theme,

  TabIndicator_thickness: 2,

  ...baseTheme
});

export const tabListTheme: TabListThemeFn = (baseTheme) => ({
  TabList_gutterHorizontal: baseTheme.space_inline_md,
  TabList_gutterVertical: baseTheme.space_stack_sm,

  TabListArrow_color: baseTheme.color_gray_80,
  TabListArrow_color_hover: baseTheme.color_theme,

  TabListOverflowContainer_boxShadowColor: baseTheme.color_white,

  ...baseTheme
});

export const tabPanelTheme: TabPanelThemeFn = (baseTheme) => ({
  // This TabList theme variable is defined here because the border is visually
  // associated with TabList, but for styling reasons, it is applied to TabPanel
  TabList_border: `1px solid ${baseTheme.borderColor}`,

  TabPanel_gap: baseTheme.space_inline_md,

  ...baseTheme
});
