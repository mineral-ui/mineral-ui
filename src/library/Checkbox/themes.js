/* @flow */
import { mapComponentThemes } from '../themes';
import { choiceTheme, choiceGroupTheme } from '../Choice';

import type { CheckboxThemeFn, CheckboxGroupThemeFn } from './types';

import { pxToEm } from '../styles';

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxTheme: CheckboxThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'Choice',
      theme: choiceTheme(baseTheme)
    },
    {
      name: 'Checkbox',
      theme: {
        CheckboxControl_backgroundColor_checked: baseTheme.checkbox_backgroundColor_checked,
        CheckboxControl_backgroundColor_checkedHover: baseTheme.checkbox_backgroundColor_checkedHover,
        CheckboxControl_borderColor: baseTheme.checkbox_borderColor,
        CheckboxControl_borderColor_checked: baseTheme.checkbox_borderColor_checked,
        CheckboxControl_borderColor_checkedHover: baseTheme.checkbox_borderColor_checkedHover,
        CheckboxControl_borderColor_hover: baseTheme.checkbox_borderColor_hover,
        CheckboxControl_borderRadius: baseTheme.borderRadius_2,
        CheckboxControl_borderWidth: `${baseTheme.checkbox_borderWidth}px`,
        CheckboxControl_size: pxToEm(20),
        CheckboxText_marginHorizontal: baseTheme.space_inline_sm
      }
    },
    baseTheme
  );

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxGroupTheme: CheckboxGroupThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'ChoiceGroup',
      theme: choiceGroupTheme(baseTheme)
    },
    {
      name: 'CheckboxGroup',
      theme: {}
    },
    baseTheme
  );
