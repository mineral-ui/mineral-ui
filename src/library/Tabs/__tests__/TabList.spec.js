/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import TabList, { componentTheme } from '../TabList';
import testThemeOverrides from '../../../../utils/testThemeOverrides';
import { mountInThemeProvider } from '../../../../utils/enzymeUtils';
import { getProcessedComponentThemeKeys } from '../../themes/processComponentTheme';

/*
 * [1] This getter does not exist on the HTMLElement.prototype in JSDOM, so we
 *     must mock it on the global. `clientWidth` and `offsetLeft`, also used by
 *     TabList always return 0, which is conveniently useful.
 * [2] Then we must clean up our global mocking in [1]
 */

function shallowTabList() {
  return shallow(<TabList />);
}

describe('TabList', () => {
  it('renders', () => {
    const tabs = shallowTabList();

    expect(tabs.exists()).toEqual(true);
  });

  describe('theme overrides', () => {
    testThemeOverrides(
      <TabList />,
      getProcessedComponentThemeKeys(componentTheme, {
        excludeKeys: [
          'TabList_gutterVertical',
          'TabListArrow_color',
          'TabListArrow_color_hover',
          'TabListOverflowContainer_boxShadowColor'
        ]
      })
    );
  });

  it('does not apply shadow when not scrollable', () => {
    const [, tabList] = mountInThemeProvider(<TabList />);

    expect(tabList).toMatchSnapshot();
  });

  it('applies shadow when scrollable', () => {
    // [1]
    // $FlowFixMe
    Object.defineProperties(window.HTMLElement.prototype, {
      scrollWidth: {
        // Note that this mocked getter is designed to always trigger overflow
        // because it's greater than offsetLeft + offsetWidth
        get: () => 10,
        configurable: true
      }
    });

    const [, tabList] = mountInThemeProvider(<TabList />);

    // [2]
    // $FlowFixMe
    Object.defineProperties(window.HTMLElement.prototype, {
      scrollWidth: {
        get: undefined
      }
    });

    expect(tabList).toMatchSnapshot();
  });
});
