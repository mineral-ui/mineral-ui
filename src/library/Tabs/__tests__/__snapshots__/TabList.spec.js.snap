// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabList applies shadow when scrollable 1`] = `
.emotion-25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: relative;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  list-style: none;
  margin: 0;
  padding: 0;
  white-space: nowrap;
}

.emotion-7 > *:not(:last-child) {
  margin-right: 1rem;
}

.emotion-3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-3:hover [role="img"] {
  color: #3272d9;
}

.emotion-3 > span {
  height: 100%;
}

.emotion-2 {
  box-sizing: border-box;
  color: #3272d9;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1rem;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  background-color: transparent;
  border-color: transparent;
  border-radius: 0;
  border-style: solid;
  border-width: 0px;
  cursor: pointer;
  display: inline-block;
  font-weight: 600;
  height: 2rem;
  margin: 0;
  min-width: 2rem;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-2 *,
.emotion-2 *::before,
.emotion-2 *::after {
  box-sizing: inherit;
}

.emotion-2:focus {
  border-color: #1d5bbf;
  color: #3272d9;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2:hover {
  color: #3272d9;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2:focus:active,
.emotion-2:focus:hover {
  border-color: #1d5bbf;
}

.emotion-2:active {
  color: #3272d9;
}

.emotion-2::-moz-focus-inner {
  border: 0;
}

.emotion-2 [role="img"] {
  box-sizing: content-box;
  color: #58606e;
  display: block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-2 [role="img"]:first-child {
  margin-right: 0.5rem;
}

.emotion-2 [role="img"]:last-child {
  margin-left: 0.5rem;
}

.emotion-2 [role="img"]:only-child {
  margin: 0;
}

.emotion-2:hover [role="img"] {
  color: #3272d9;
}

.emotion-2 > span {
  height: 100%;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-height: 100%;
  pointer-events: none;
  width: 100%;
}

.emotion-0 {
  fill: currentcolor;
  font-size: 1rem;
  height: 1.25rem;
  width: 1.25rem;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 0%;
}

.emotion-14::before {
  bottom: 1px;
}

.emotion-9 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: hidden;
  padding-left: 2px;
  padding-right: 2px;
}

.emotion-8 {
  outline: 0;
  overflow-x: auto;
  -webkit-transition: outline 0.1s 0.25s;
  transition: outline 0.1s 0.25s;
  overflow: -moz-scrollbars-none;
  -ms-overflow-style: none;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: hidden;
  padding-left: 2px;
  padding-right: 2px;
}

.emotion-8:focus {
  outline: 1px solid #1d5bbf;
}

.emotion-8::-webkit-scrollbar {
  display: none;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 0%;
}

.emotion-13::before {
  bottom: 0;
  box-shadow: inset -24px 0 22px -14px #ffffff;
  content: "";
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
}

.emotion-13::before {
  bottom: 1px;
}

<TabList
  position="top"
>
  <TabList
    innerRef={[Function]}
  >
    <div
      className="emotion-25"
    >
      <IncrementButton
        icon={<IconChevronLeft />}
        key="start"
        onClick={[Function]}
      >
        <ArrowButton
          aria-hidden={true}
          iconStart={<IconChevronLeft />}
          minimal={true}
          onClick={[Function]}
          size="medium"
          tabIndex={-1}
        >
          <ArrowButton
            aria-hidden={true}
            className="emotion-3"
            iconStart={<IconChevronLeft />}
            minimal={true}
            onClick={[Function]}
            size="medium"
            tabIndex={-1}
          >
            <Themed(Button)
              aria-hidden={true}
              className="emotion-3"
              iconStart={<IconChevronLeft />}
              minimal={true}
              onClick={[Function]}
              size="medium"
              tabIndex={-1}
            >
              <ThemeProvider>
                <Fragment>
                  <DocumentFontSizeProvider>
                    <ThemeProvider>
                      <Button
                        aria-hidden={true}
                        className="emotion-3"
                        element="button"
                        iconStart={<IconChevronLeft />}
                        minimal={true}
                        onClick={[Function]}
                        size="medium"
                        tabIndex={-1}
                        type="button"
                      >
                        <Button
                          aria-hidden={true}
                          className="emotion-3"
                          element="button"
                          minimal={true}
                          onClick={[Function]}
                          size="medium"
                          tabIndex={-1}
                          type="button"
                        >
                          <button
                            aria-hidden={true}
                            className="emotion-2"
                            onClick={[Function]}
                            tabIndex={-1}
                            type="button"
                          >
                            <Styled(span)>
                              <span
                                className="emotion-1"
                              >
                                <IconChevronLeft
                                  key="iconStart"
                                  size="medium"
                                >
                                  <Icon
                                    rtl={true}
                                    size="medium"
                                  >
                                    <Styled(svg)
                                      aria-hidden={true}
                                      focusable="false"
                                      role="img"
                                      rtl={true}
                                      size="medium"
                                      viewBox="0 0 24 24"
                                    >
                                      <svg
                                        aria-hidden={true}
                                        className="emotion-0"
                                        focusable="false"
                                        role="img"
                                        viewBox="0 0 24 24"
                                      >
                                        <g>
                                          <path
                                            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                                          />
                                        </g>
                                      </svg>
                                    </Styled(svg)>
                                  </Icon>
                                </IconChevronLeft>
                              </span>
                            </Styled(span)>
                          </button>
                        </Button>
                      </Button>
                    </ThemeProvider>
                  </DocumentFontSizeProvider>
                  <EventListener
                    listeners={
                      Array [
                        Object {
                          "event": "DOMContentLoaded",
                          "handler": [Function],
                          "target": "document",
                        },
                      ]
                    }
                  />
                </Fragment>
              </ThemeProvider>
            </Themed(Button)>
          </ArrowButton>
        </ArrowButton>
      </IncrementButton>
      <withProps(Inner)
        key="inner"
        position="top"
        scrollX={true}
      >
        <Inner
          hideScrollbars={true}
          position="top"
          scrollX={true}
          tabIndex={null}
        >
          <Inner
            className="emotion-14"
            hideScrollbars={true}
            position="top"
            scrollX={true}
            tabIndex={null}
          >
            <Themed(OverflowContainerWithShadows)
              className="emotion-14"
              hideScrollbars={true}
              position="top"
              scrollX={true}
              tabIndex={null}
            >
              <ThemeProvider>
                <Fragment>
                  <DocumentFontSizeProvider>
                    <ThemeProvider>
                      <OverflowContainerWithShadows
                        className="emotion-14"
                        hideScrollbars={true}
                        position="top"
                        scrollX={true}
                        tabIndex={null}
                      >
                        <OverflowContainerWithShadows
                          className="emotion-14"
                          hasShadowLeft={false}
                          hasShadowRight={true}
                          position="top"
                          tabIndex={null}
                        >
                          <div
                            className="emotion-13"
                            tabIndex={null}
                          >
                            <Scroller
                              containerRef={[Function]}
                              hideScrollbars={true}
                              onScroll={[Function]}
                              scrollX={true}
                              tabIndex={0}
                            >
                              <Scroller
                                className="emotion-9"
                                containerRef={[Function]}
                                hideScrollbars={true}
                                onScroll={[Function]}
                                scrollX={true}
                                tabIndex={0}
                              >
                                <Themed(OverflowContainer)
                                  className="emotion-9"
                                  containerRef={[Function]}
                                  hideScrollbars={true}
                                  onScroll={[Function]}
                                  scrollX={true}
                                  tabIndex={0}
                                >
                                  <ThemeProvider>
                                    <Fragment>
                                      <DocumentFontSizeProvider>
                                        <ThemeProvider>
                                          <OverflowContainer
                                            className="emotion-9"
                                            containerRef={[Function]}
                                            hideScrollbars={true}
                                            onScroll={[Function]}
                                            scrollX={true}
                                            tabIndex={0}
                                          >
                                            <OverflowContainer
                                              className="emotion-9"
                                              containerRef={[Function]}
                                              hideScrollbars={true}
                                              innerRef={[Function]}
                                              onScroll={[Function]}
                                              scrollX={true}
                                              tabIndex={0}
                                            >
                                              <div
                                                className="emotion-8"
                                                onScroll={[Function]}
                                                tabIndex={0}
                                              >
                                                <Styled(ul)
                                                  count={0}
                                                  innerRef={[Function]}
                                                >
                                                  <ul
                                                    className="emotion-7"
                                                  />
                                                </Styled(ul)>
                                                <EventListener
                                                  listeners={
                                                    Array [
                                                      Object {
                                                        "event": "resize",
                                                        "handler": [Function],
                                                        "target": "window",
                                                      },
                                                    ]
                                                  }
                                                />
                                              </div>
                                            </OverflowContainer>
                                          </OverflowContainer>
                                        </ThemeProvider>
                                      </DocumentFontSizeProvider>
                                      <EventListener
                                        listeners={
                                          Array [
                                            Object {
                                              "event": "DOMContentLoaded",
                                              "handler": [Function],
                                              "target": "document",
                                            },
                                          ]
                                        }
                                      />
                                    </Fragment>
                                  </ThemeProvider>
                                </Themed(OverflowContainer)>
                              </Scroller>
                            </Scroller>
                          </div>
                        </OverflowContainerWithShadows>
                      </OverflowContainerWithShadows>
                    </ThemeProvider>
                  </DocumentFontSizeProvider>
                  <EventListener
                    listeners={
                      Array [
                        Object {
                          "event": "DOMContentLoaded",
                          "handler": [Function],
                          "target": "document",
                        },
                      ]
                    }
                  />
                </Fragment>
              </ThemeProvider>
            </Themed(OverflowContainerWithShadows)>
          </Inner>
        </Inner>
      </withProps(Inner)>
      <IncrementButton
        icon={<IconChevronRight />}
        key="end"
        onClick={[Function]}
      >
        <ArrowButton
          aria-hidden={true}
          iconStart={<IconChevronRight />}
          minimal={true}
          onClick={[Function]}
          size="medium"
          tabIndex={-1}
        >
          <ArrowButton
            aria-hidden={true}
            className="emotion-3"
            iconStart={<IconChevronRight />}
            minimal={true}
            onClick={[Function]}
            size="medium"
            tabIndex={-1}
          >
            <Themed(Button)
              aria-hidden={true}
              className="emotion-3"
              iconStart={<IconChevronRight />}
              minimal={true}
              onClick={[Function]}
              size="medium"
              tabIndex={-1}
            >
              <ThemeProvider>
                <Fragment>
                  <DocumentFontSizeProvider>
                    <ThemeProvider>
                      <Button
                        aria-hidden={true}
                        className="emotion-3"
                        element="button"
                        iconStart={<IconChevronRight />}
                        minimal={true}
                        onClick={[Function]}
                        size="medium"
                        tabIndex={-1}
                        type="button"
                      >
                        <Button
                          aria-hidden={true}
                          className="emotion-3"
                          element="button"
                          minimal={true}
                          onClick={[Function]}
                          size="medium"
                          tabIndex={-1}
                          type="button"
                        >
                          <button
                            aria-hidden={true}
                            className="emotion-2"
                            onClick={[Function]}
                            tabIndex={-1}
                            type="button"
                          >
                            <Styled(span)>
                              <span
                                className="emotion-1"
                              >
                                <IconChevronRight
                                  key="iconStart"
                                  size="medium"
                                >
                                  <Icon
                                    rtl={true}
                                    size="medium"
                                  >
                                    <Styled(svg)
                                      aria-hidden={true}
                                      focusable="false"
                                      role="img"
                                      rtl={true}
                                      size="medium"
                                      viewBox="0 0 24 24"
                                    >
                                      <svg
                                        aria-hidden={true}
                                        className="emotion-0"
                                        focusable="false"
                                        role="img"
                                        viewBox="0 0 24 24"
                                      >
                                        <g>
                                          <path
                                            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                                          />
                                        </g>
                                      </svg>
                                    </Styled(svg)>
                                  </Icon>
                                </IconChevronRight>
                              </span>
                            </Styled(span)>
                          </button>
                        </Button>
                      </Button>
                    </ThemeProvider>
                  </DocumentFontSizeProvider>
                  <EventListener
                    listeners={
                      Array [
                        Object {
                          "event": "DOMContentLoaded",
                          "handler": [Function],
                          "target": "document",
                        },
                      ]
                    }
                  />
                </Fragment>
              </ThemeProvider>
            </Themed(Button)>
          </ArrowButton>
        </ArrowButton>
      </IncrementButton>
      <EventListener
        listeners={
          Array [
            Object {
              "event": "resize",
              "handler": [Function],
              "target": "window",
            },
          ]
        }
      />
    </div>
  </TabList>
</TabList>
`;

exports[`TabList does not apply shadow when not scrollable 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: relative;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  list-style: none;
  margin: 0;
  padding: 0;
  white-space: nowrap;
}

.emotion-0 > *:not(:last-child) {
  margin-right: 1rem;
}

<TabList
  position="top"
>
  <TabList
    innerRef={[Function]}
  >
    <div
      className="emotion-1"
    >
      <Styled(ul)
        count={0}
        innerRef={[Function]}
      >
        <ul
          className="emotion-0"
        />
      </Styled(ul)>
      <EventListener
        listeners={
          Array [
            Object {
              "event": "resize",
              "handler": [Function],
              "target": "window",
            },
          ]
        }
      />
    </div>
  </TabList>
</TabList>
`;
